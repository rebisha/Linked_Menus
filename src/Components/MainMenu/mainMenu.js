import React, { useState, Fragment } from "react";
// components
import Button from "../Button/button";
import Form from "../Form/form";
import LinkedMenu from "../LinkedMenu/linkedMenu";

const MainMenu = ({ item }) => {
  const { id, name, hasForm, subMenu } = item;

  const [formToggle, setFormToggle] = useState(false);
  const [headingToggle, setHeadingToggle] = useState(false);
  const [subMenuToggle, setSubMenuToggle] = useState(false);

  const handleClick = e => {
    setFormToggle(hasForm);
    setHeadingToggle(!headingToggle);
    setSubMenuToggle(!subMenuToggle);
  };

  const handleCancel = e => {
    setFormToggle(!formToggle);
  };

  return (
    <Fragment>
      <Button
        type="button"
        className="sub-menu-item"
        name={name}
        onClick={handleClick}
      />

      {formToggle && <Form handleCancel={handleCancel} />}
      {headingToggle && !hasForm && !subMenu && (
        <h1>This functionality hasn't been implemented yet!!</h1>
      )}
      {subMenu &&
        subMenuToggle &&
        subMenu.map(
          item =>
            item && (
              <LinkedMenu
                key={item.id}
                subMenu={item}
                formToggle={formToggle}
                headingToggle={headingToggle}
                handleCancel={handleCancel}
              />
            )
        )}
    </Fragment>
  );
};

export default MainMenu;
