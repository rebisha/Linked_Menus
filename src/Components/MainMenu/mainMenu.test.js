import React from "react";
import MainMenu from "./mainMenu";
import renderer from "react-test-renderer";

const item = [
  {
    name: "Example 1",
    hasForm: false,
    id: 0
  },
  {
    name: "Example 2",
    hasForm: false,
    id: 1
  },
  {
    name: "Example 3",
    hasForm: true,
    id: 2,
    subMenu: [
      {
        id: 3,
        name: "SubMenu 1",
        hasForm: false
      },
      {
        id: 4,
        name: "SubMenu 1",
        hasForm: false
      }
    ]
  }
];

test("MainMenu component renders correctly", () => {
  const component = renderer.create(<MainMenu item={item} />);
  expect(component).toMatchSnapshot();
});
