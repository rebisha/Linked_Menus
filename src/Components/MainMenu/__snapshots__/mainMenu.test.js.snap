// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu component renders correctly 1`] = `
<div
  className="main-menu"
>
  <button
    className="button-wrapper"
    onClick={[Function]}
    type="button"
  />
</div>
`;
